smart_open/tests/test_smart_open.py

```diff
@@ -483,6 +483,40 @@ def test_s3_mode_mock(self, mock_write, mock_boto):
         mock_boto.connect_s3().get_bucket.assert_called_with("mybucket")
```

Comment:
OK, added a test which makes sure that the 'port' argument isn't even passed in the default case.